### ISSUE TITLE
'Model' object has no attribute 'total_loss' [2.2 Eager]

### ISSUE THREAD

**Issue by OverLordGoldDragon** [tensorflow/tensorflow#39776]
Below works in TF 2.1 and 2.0 Eager & Graph, and 2.2 Graph, but not 2.2 Eager; how, then, are we to get gradients in 2.2 Eager?

<hr>

```python
import numpy as np
import tensorflow as tf
from tensorflow.keras.layers import Input, Dense
from tensorflow.keras.models import Model
from tensorflow.keras import backend as K

ipt = Input((16,))
out = Dense(16)(ipt)
model = Model(ipt, out)
model.compile('adam', 'mse')

x = y = np.random.randn(32, 16)
model.train_on_batch(x, y)

outputs = model.optimizer.get_gradients(model.total_loss, model.trainable_weights)
inputs  = [model.inputs[0], model._feed_targets[0]]

grads_fn = K.function(inputs, outputs)
grads    = grads_fn([x, y])
_ = [print(g.shape) for g in grads]
```

**Comment by OverLordGoldDragon** [comment]
[Figured it out](https://stackoverflow.com/questions/61887944/how-to-get-gradients-in-tf-2-2-eager/#answer-61952452)
